src/Workspaces/CSharp/Portable/EmbeddedLanguages/LanguageServices/CSharpEmbeddedLanguageProvider.cs

```diff
@@ -0,0 +1,24 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Composition;
+using Microsoft.CodeAnalysis.CSharp.EmbeddedLanguages.VirtualChars;
+using Microsoft.CodeAnalysis.EmbeddedLanguages.LanguageServices;
+using Microsoft.CodeAnalysis.Host.Mef;
+
+namespace Microsoft.CodeAnalysis.CSharp.EmbeddedLanguages.LanguageServices
+{
+    [ExportLanguageService(typeof(IEmbeddedLanguageProvider), LanguageNames.CSharp), Shared]
+    internal class CSharpEmbeddedLanguageProvider : AbstractEmbeddedLanguageProvider
```

Comment:
>CSharpEmbeddedLanguageProvider [](start = 19, length = 30)

Are you expecting language-specific embedded languages?
If not, can we remove `CSharpEmbeddedLanguageProvider` and `VisualBasicEmbeddedLanguageProvider` and make `AbstractEmbeddedLanguageProvider` concrete and exported?

Reply:
Ah. Never mind :-) Thanks

---
In reply to: [208754259](https://github.com/dotnet/roslyn/pull/28927#discussion_r208754259) [](ancestors = 208754259)